src/matchers/__tests__/to-have-prop.test.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Button, Text, View } from 'react-native';
+import { render } from '../..';
+import '../extend-expect';
+
+test('.toHaveProp', () => {
+  const { queryByTestId } = render(
+    <View style={null} testID="view">
+      <Text allowFontScaling={false} testID="text" ellipsizeMode="head">
+        text
+      </Text>
+      <Button disabled testID="button" title="ok" />
+    </View>
+  );
+
+  expect(queryByTestId('button')).toHaveProp('accessibilityState', {
+    disabled: true,
+  });
+  expect(queryByTestId('text')).toHaveProp('ellipsizeMode', 'head');
+  expect(queryByTestId('text')).toHaveProp('allowFontScaling', false);
+
+  expect(queryByTestId('button')).not.toHaveProp('accessibilityStates');
+  expect(queryByTestId('button')).not.toHaveProp('ellipsizeMode', undefined);
+  expect(queryByTestId('button')).not.toHaveProp('allowFontScaling', false);
+  expect(queryByTestId('text')).not.toHaveProp('style');
+
+  // title is no longer findable as it is a React child
+  expect(() =>
```

Comment:
I guess it was like that in jest-native matcher but wouldn't it be better to use not rather than expect the contrary to throw?

Reply:
Oh I realize now using this could allow to test the message, can you replace to throw by toThrowErrorMatchingInlineSnapshot()?